src/main/java/org/folio/mosaic/service/MosaicOrderConverter.java

```diff
@@ -104,207 +165,255 @@ private void applyOverrides(CompositePurchaseOrder order, MosaicOrder mosaicOrde
     if (mosaicOrder.getShipTo() != null) {
       order.setShipTo(mosaicOrder.getShipTo());
     }
-
     if (CollectionUtils.isNotEmpty(mosaicOrder.getAcqUnitIds())) {
       order.setAcqUnitIds(mosaicOrder.getAcqUnitIds());
     }
-
     if (mosaicOrder.getCustomFields() != null) {
       var convertedCustomFields = new CustomFields();
       mosaicOrder.getCustomFields().getAdditionalProperties().forEach(convertedCustomFields::withAdditionalProperty);
       order.setCustomFields(convertedCustomFields);
     }
 
-    var poLine = new PoLine()
-      .withId(UUID.randomUUID().toString())
-      .withSource(PoLine.Source.API);
+    applyOverridesToPoLine(order, mosaicOrder);
+  }
 
+  /**
+   * Applies overrides from the request to the poKine.
```

Comment:
some typo here in java docs.
Our converter class is so big now, propose to introduce 2 converters: one for order and another for po line with single interface methods like createFromTemplate, applyOverrides